src/lib.rs

```diff
@@ -297,9 +371,38 @@ pub trait Backend: fmt::Debug {
     fn set_prefix(&mut self, prefix: &str);
 }
 
-pub fn read_lock_file<P: AsRef<Path>>(lock_path: P) -> Result<Vec<Krate>, Error> {
+pub fn read_cargo_config(config_path: &Path) -> Result<HashMap<String, Registry>, Error> {
```

Comment:
I'd prefer just a `Vec<>` here, HashMap is a big overkill for what is most likely going to be 1 custom registry.